retnet/modeling_retnet.py

```diff
@@ -636,7 +636,8 @@ def generate(
                            use_cache=True,
                            return_dict=True,
                            sequence_offset=prompt_len + i)
-            logit = outputs.logits
+            logit = outputs.logits.squeeze(0)
```

Comment:
This is not the correct way to go. The logits shape should be `[batch_size, seq_len, vocab_size]`, so if the batch size is not 1, this line does nothing.

Instead, we should take the last token's logit only, by setting

```
logit = outputs.logits[:, -1]  # [batch_size, vocab_size]
```

This will solve the problem.

Reply:
Right, I committed a change, using outputs.logits[:, -1, :] to make it a bit more explicit.